### ISSUE TITLE
Upstream proxy is not solving youtube's problem

### ISSUE THREAD

**Issue by rti2** [msasanmh/DNSveil#220]
https://github.com/msasanmh/DNSveil/blob/main/Help/YoutubePlaybackFix_ForIran.md
It isn't working. Not with oblivion nor my own v2ray (both can play youtube with no problem when used primary)
what should I do? 
(I would comment one of the proxies and test the other)

 `// Variables
SmartDns1 = https://free.shecan.ir/dns-query;
SmartDns2 = https://dns.403.online/dns-query;
SmartDns3 = https://185.55.226.26/dns-query;
v2ray = http://127.0.0.1:10809;
OblivionProxy = http://127.0.0.1:8086;
blockport:53,80;
// YouTube
youtube.com|dnsdomain:google.com;sni:google.com;
ytimg.com|dnsdomain:google.com;
*.ytimg.com|dnsdomain:google.com;
ggpht.com|dnsdomain:google.com;
*.ggpht.com|dnsdomain:*.googleusercontent.com;
*.googleapis|dnsdomain:google.com;
*.googlevideo.com|dnsdomain:*.c.docs.google.com;sni:google.com;
// Youtube Playback Issue
//*.googleapis.com|proxy:v2ray;
*.googleapis.com|proxy:OblivionProxy;
// Use Smart DNS For These Domains
developers.google.com|--;dns:SmartDns1,SmartDns2,SmartDns3;
*.googleusercontent.com|--;dns:SmartDns1,SmartDns2,SmartDns3;
developer.android.com|--;dns:SmartDns1,SmartDns2,SmartDns3;
gemini.google.com|--;dns:SmartDns1,SmartDns2,SmartDns3;
*.openai.com|--;dns:SmartDns1;
claude.ai|--;dns:SmartDns1,SmartDns2,SmartDns3;
*.claude.ai|--;dns:SmartDns1,SmartDns2,SmartDns3;
spotify.com|--;dns:SmartDns1,SmartDns2,SmartDns3;
*.spotify.com|--;dns:SmartDns1,SmartDns2,SmartDns3;
// Don't Apply DPI Bypass To These Domains
google.com|--;
*.google.com|--;
github.com|--;
*.github.com|--;
githubusercontent.com|--;
*.githubusercontent.com|--;
stackoverflow.com|--;
*.stackoverflow.com|--;
*.sstatic.net|--;
*.cookielaw.org|--;
every1dns.com|--;
*.every1dns.com|--;
nslookup.io|--;
*.nslookup.io|--;
php.net|--;
save.tube|--;
// Apply Defaults To Other Domains
*|+;`

**Comment by msasanmh** [comment]
This line is incorrect. Remove it:
`*.googleapis|dnsdomain:google.com;`

Consider these:
1. Try Oblivion on Warp On Warp.
2. In your V2ray client, make sure it's not disabling advertisements. 
3. If you have an ad blocker extension on your browser, disable it for youtube.com domain.
4. After modifying Rules, disconnect all and reconnect. 
5. Make sure you're using SDC's Proxy, not GoodbyeDPI. 
6. In Rules, apply proxy to all youtube related domains, except `*.googlevideo.com`

If the issue persist, you have to wait for SDC Plus (DNSveil).
Unfortunately, it takes at least 6 months to get ready. I'm just one person after all, not a team.

Your English is good and you know what exactly this issue is, so you can create a post on YouTube community center to let them know about their ridiculous issue.

**Comment by msasanmh** [comment]
> https://github.com/msasanmh/DNSveil/blob/main/Help/YoutubePlaybackFix_ForIran.md
> It isn't working. Not with oblivion nor my own v2ray (both can play youtube with no problem when used primary)
> what should I do? 
> (I would comment one of the proxies and test the other)
> 
>  `// Variables
> SmartDns1 = https://free.shecan.ir/dns-query;
> SmartDns2 = https://dns.403.online/dns-query;
> SmartDns3 = https://185.55.226.26/dns-query;
> v2ray = http://127.0.0.1:10809;
> OblivionProxy = http://127.0.0.1:8086;
> blockport:53,80;
> // YouTube
> youtube.com|dnsdomain:google.com;sni:google.com;
> ytimg.com|dnsdomain:google.com;
> *.ytimg.com|dnsdomain:google.com;
> ggpht.com|dnsdomain:google.com;
> *.ggpht.com|dnsdomain:*.googleusercontent.com;
> *.googleapis|dnsdomain:google.com;
> *.googlevideo.com|dnsdomain:*.c.docs.google.com;sni:google.com;
> // Youtube Playback Issue
> //*.googleapis.com|proxy:v2ray;
> *.googleapis.com|proxy:OblivionProxy;
> // Use Smart DNS For These Domains
> developers.google.com|--;dns:SmartDns1,SmartDns2,SmartDns3;
> *.googleusercontent.com|--;dns:SmartDns1,SmartDns2,SmartDns3;
> developer.android.com|--;dns:SmartDns1,SmartDns2,SmartDns3;
> gemini.google.com|--;dns:SmartDns1,SmartDns2,SmartDns3;
> *.openai.com|--;dns:SmartDns1;
> claude.ai|--;dns:SmartDns1,SmartDns2,SmartDns3;
> *.claude.ai|--;dns:SmartDns1,SmartDns2,SmartDns3;
> spotify.com|--;dns:SmartDns1,SmartDns2,SmartDns3;
> *.spotify.com|--;dns:SmartDns1,SmartDns2,SmartDns3;
> // Don't Apply DPI Bypass To These Domains
> google.com|--;
> *.google.com|--;
> github.com|--;
> *.github.com|--;
> githubusercontent.com|--;
> *.githubusercontent.com|--;
> stackoverflow.com|--;
> *.stackoverflow.com|--;
> *.sstatic.net|--;
> *.cookielaw.org|--;
> every1dns.com|--;
> *.every1dns.com|--;
> nslookup.io|--;
> *.nslookup.io|--;
> php.net|--;
> save.tube|--;
> // Apply Defaults To Other Domains
> *|+;`

You're right it didn't work with HTTP, I changed it to SOCKS5 and it start working. 
Please change your rules to this:

`OblivionProxy = socks5://127.0.0.1:8086;`
`v2ray = socks5://127.0.0.1:10808;`

Let me know if it works.

**Comment by msasanmh** [comment]
I found out another trick. check this out again:
https://github.com/msasanmh/DNSveil/blob/main/Help/YoutubePlaybackFix_ForIran.md

**Comment by rti2** [comment]
> This line is incorrect. Remove it: `*.googleapis|dnsdomain:google.com;`
> 
> Consider these:
> 
> 1. Try Oblivion on Warp On Warp.
> 2. In your V2ray client, make sure it's not disabling advertisements.
> 3. If you have an ad blocker extension on your browser, disable it for youtube.com domain.
> 4. After modifying Rules, disconnect all and reconnect.
> 5. Make sure you're using SDC's Proxy, not GoodbyeDPI.
> 6. In Rules, apply proxy to all youtube related domains, except `*.googlevideo.com`
> 
> If the issue persist, you have to wait for SDC Plus (DNSveil). Unfortunately, it takes at least 6 months to get ready. I'm just one person after all, not a team.
> 
> Your English is good and you know what exactly this issue is, so you can create a post on YouTube community center to let them know about their ridiculous issue.

I made a post on YT's community , ولی به یورشون هم نمی گیرن

**Comment by rti2** [comment]
> I found out another trick. check this out again: https://github.com/msasanmh/DNSveil/blob/main/Help/YoutubePlaybackFix_ForIran.md

https://github.com/rti2/embed-redirect
its an extension which redirects yt links to embed ones

**Comment by msasanmh** [comment]
> > I found out another trick. check this out again: https://github.com/msasanmh/DNSveil/blob/main/Help/YoutubePlaybackFix_ForIran.md
> 
> https://github.com/rti2/embed-redirect
> its an extension which redirects yt links to embed ones 

Awesome. Thanks for sharing.
